data/tmdb-movies/src/main/java/io/fimltime/data/tmdb/movies/TmdbMovieRepositoryImpl.kt

```diff
@@ -39,21 +45,25 @@ internal class TmdbMovieRepositoryImpl @Inject constructor(
                 )
               }
 
-              is Result.Success -> result.run {
-                copy(
-                  data = data.copy(
-                    ids = data.ids.copy(
-                      traktId = traktId,
+              is Result.Success -> {
+                result.run {
+                  copy(
+                    data = data.copy(
+                      ids = data.ids.copy(
+                        traktId = traktId,
+                      ),
+                      isWatched = watched.data,
                     ),
-                    isWatched = watched.data,
-                  ),
-                )
+                  )
+                }
               }
             }
           }
         }
       }
     }
+
+    emit(result)
```

Comment:
We are not implementing offline first correctly here since you are always emitting the result of network call. We should only emit the network call result if local movie is null, also no need to even emit the success result if we observe the `getMovieByTmdbId`

Reply:
I want to store some permanent data of movies like title or description to show very quickly in UI and when Network result arrived I update my database and emit new data.
So if we want to implement it as offline first do you suggest store all details? and in ```GetMovieDetailsUseCase``` observe on database and only request network call in ```TmdbMovieRepository```?

Reply:
If you want to use cache for this purpose we have to think more about the implementation.

**scenario:** 
You have a cached data for a movie. User navigates to a movie detail screen again in since we have the cache, we quickly show him the initial title, image and description. But after a few seconds the network call fails. What's your action now? 
**Option A:** 
It's not a good idea (UX) to show an error screen because a few moments ago you were displaying the initial data and suddenly you switch to error screen. 
**Option B**
Show a snack bar or toast indicating that network failed. So what is next for user? is the info on screen enough for him to use or should he refresh the screen? Do you want to add a load again or refresh button in every placed with missing data? (genres or langs, ...)

That's why I suggest to store all data to make the implementation as simple as possible.

You can always observe the database and only propagate the network error result if there is no cached data.